### ISSUE TITLE
Adjacent and direct descendant selectors are not space separated

### ISSUE THREAD

**Issue by raduluchian** [beautify-web/js-beautify#676]
Altought the CSS rule for "selector_separator" is set to " " (space character) this doesn't apply to rules that use direct descendant selectors or adjacent selectors. For example the following code...

    a+b {
        ...
    }

    a>b {
        ...
    }
... doesn't get formatted into this:

    a + b {
         ...
    }

    a > b {
        ...
    }

**Comment by bitwiseman** [comment]
So, the space shouldn't be added, but is?

**Comment by raduluchian** [comment]
Nope, quite the opposite: the space should be added but isn't.

**Comment by bitwiseman** [comment]
Oh, okay.  :smile: 
What would you expect to see if "selector_separator" were set to "\n"?

**Comment by raduluchian** [comment]
I suppose to see each selector on a separate line, even if they are part of the same rule. So I would expect this code:

    .one .two .three {
        ...
    }

to be formatted into this:

    .one
    .two
    .three {
        ...
    }

**Comment by bitwiseman** [comment]
What about your original example?

**Comment by raduluchian** [comment]
a>b {
        ...
    }

would turn into this:

    a
    >
    b {
        ...
    }

**Comment by raduluchian** [comment]
This issue had it's one year birthday yesterday. :) Any updates on it?

**Comment by bitwiseman** [comment]
Nope. This project is widely used and I devote what time I can to it, but it needs more contributors.

**Comment by bitwiseman** [comment]
@raduluchian - Have you tried setting `space_around_selector_separator = true`? 
Think that will do what you want.

**Comment by smonff** [comment]
@bitwiseman The `space_around_selector_separator` parameter set to `true` fix this issue.

Though, it should be like that by default: I cannot think of a situation where you wouldn't want js-beautify-css not to add (or maintain existing spaces) between `+` or `>` relationship separators because it breaks your CSS rules.

**Comment by FrEaKmAn** [comment]
For anyone wondering, this has been update to `space_around_combinator`

**Comment by dosicker** [comment]
> For anyone wondering, this has been update to `space_around_combinator`

Woo woo woo ~ tks tks tks very!

**Comment by dosicker** [comment]
> For anyone wondering, this has been update to `space_around_combinator`

Sob~ this property configuration still doesn't work...
Then I found this related update adjustment [Implement option to support adding spaces between CSS selectors](https://github.com/beautifier/js-beautify/pull/2029)

**Comment by bitwiseman** [comment]
@dosicker 
Could you open a new issue for this.
I think it would make sense if one of the beautifier modules sees a JSON element matching itself it should drill into that.  
The fact that they don't is not an outright bug, but it's behavior that can be fixed.

**Comment by dosicker** [comment]
> @dosicker Could you open a new issue for this. I think it would make sense if one of the beautifier modules sees a JSON element matching itself it should drill into that. The fact that they don't is not an outright bug, but it's behavior that could be improved.

Thanks you reply!
and...Yeah, maybe you misunderstood me. My statement above was to say that I was stupid... which led to the wrong use of the property...

But what you meant by this sentence is that the configuration of this property should actually be written in the CSS object, not in the top-level configuration, that right?

**Comment by bitwiseman** [comment]
It's okay that it works in the root object, that is reasonable behavior.
It is a bug that it doesn't work when inside the `css` object.

**Comment by dosicker** [comment]
> @dosicker It's okay that it works in the root object, that is reasonable behavior. It is a bug that it doesn't work when inside the `css` object.

yeah, ok~ got it.
I'll take time to create a new issue.
